tasks/main.yml

```diff
@@ -223,6 +223,13 @@
     inventory_mode: "{{ zabbix_inventory_mode }}"
     interfaces: "{{ zabbix_agent_interfaces }}"
     visible_name: "{{ zabbix_visible_hostname|default(zabbix_agent_hostname) }}"
+    tls_accept: "{{ {'unencrypted': '1', 'psk': '2', 'cert' : '4'}[zabbix_agent_tlsaccept] | default('1') }}"
+    tls_connect: "{{ {'unencrypted': '1', 'psk': '2', 'cert' : '4'}[zabbix_agent_tlsconnect] | default('1') }}"
+    tls_psk_identity: "{{ zabbix_agent_tlspskidentity if (zabbix_agent_tlsaccept == 'psk') else '' }}"
```

Comment:
This will fail if no encryption is configured because ansible zabbix_host module doesn't want an empty string. You should use `default(omit)`